api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -687,6 +692,25 @@ func (m *AzureManagedControlPlane) validateOIDCIssuerProfileUpdate(old *AzureMan
 	return allErrs
 }
 
+// validateFleetsMember validates a FleetsMember.
+func (m *AzureManagedControlPlane) validateFleetsMember(old *AzureManagedControlPlane) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if old.Spec.FleetsMember == nil || m.Spec.FleetsMember == nil {
+		return allErrs
+	}
```

Comment:
This statement seems to allow _addition of new FleetMember to the spec_ or _deletion of FleetMember from the spec_ since `||` is a short-circuited OR, are those allowed scenarios ?

Reply:
Yes we're allowed to join an existing cluster to a Fleet, or detach it from the Fleet by removing the FleetsMember spec.